cmd/gardener-resource-manager/app/app.go

```diff
@@ -206,17 +236,33 @@ func NewControllerManagerCommand(ctx context.Context) *cobra.Command {
 			entryLog.Info("Managed resource health controller", "syncPeriod", healthSyncPeriod.String())
 			entryLog.Info("Managed resource health controller", "maxConcurrentWorkers", healthMaxConcurrentWorkers)
 
-			if err := mgr.Start(ctx.Done()); err != nil {
-				entryLog.Error(err, "error running manager")
-				os.Exit(1)
-			}
+			// start the target cache and exit if there was an error
+			go func() {
+				if err := targetCache.Start(ctx.Done()); err != nil {
+					entryLog.Error(err, "error syncing target cache")
+					os.Exit(1)
+				}
+			}()
+
+			targetCache.WaitForCacheSync(ctx.Done())
+
+			go func() {
```

Comment:
I wonder if we should synchronize this routine with the main one in order to wait for the manager to be finished. WDYT?

Reply:
Can you elaborate, what you mean by "synchronize this routine with the main one"? 
The manager actually does exactly the same as our main routine, it waits for the `ctx.Done()` channel to be closed (or an error happened) and stops afterwards.

Reply:
But we could also run `mgr.Start()` directly in the main routine (as it was before) as it blocks until the stop channel is closed, if that is what you mean.